cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TranslatorSuite.scala

```diff
@@ -0,0 +1,434 @@
+    "</div><div>This will be translated.</div>")).toDF("text")
+
+  lazy val textDf5: DataFrame = Seq(List("The word <mstrans:dictionary translation=wordomatic>word " +
+    "or phrase</mstrans:dictionary> is a dictionary entry.")).toDF("text")
+
+}
+
+class TranslateSuite extends TransformerFuzzing[Translate]
+  with TranslatorKey with Flaky with TranslatorUtils {
+
+  lazy val translate: Translate = new Translate()
+    .setSubscriptionKey(translatorKey)
+    .setLocation("eastus")
+    .setToLanguage(Seq("zh-Hans"))
+    .setTextCol("text")
+    .setOutputCol("translation")
+    .setConcurrency(5)
+
+  lazy val translate2: Translate = new Translate()
+    .setSubscriptionKey(translatorKey)
+    .setLocation("eastus")
+    .setToLanguage(Seq("zh-Hans"))
+    .setToScript("Latn")
+    .setTextCol("text")
+    .setOutputCol("translation")
+    .setConcurrency(5)
+
+  lazy val translate3: Translate = new Translate()
+    .setSubscriptionKey(translatorKey)
+    .setLocation("eastus")
+    .setFromLanguage("en")
+    .setToLanguage(Seq("fr"))
+    .setIncludeAlignment(true)
+    .setTextCol("text")
+    .setOutputCol("translation")
+    .setConcurrency(5)
+
+  lazy val translate4: Translate = new Translate()
+    .setSubscriptionKey(translatorKey)
+    .setLocation("eastus")
+    .setFromLanguage("en")
+    .setToLanguage(Seq("fr"))
+    .setIncludeSentenceLength(true)
+    .setTextCol("text")
+    .setOutputCol("translation")
+    .setConcurrency(5)
+
+  test("Translate multiple pieces of text with language autodetection") {
+    val results = translate.transform(textDf2)
+      .withColumn("translation", flatten(col("translation.translations")))
+      .withColumn("translation", col("translation.text"))
+      .select("translation").collect()
+    val headStr = results.head.getSeq(0).mkString("\n")
+    assert(headStr === "你好，你叫什么名字？\n再见")
+  }
+
+  test("Translate with transliteration") {
+    val results = translate2
+      .transform(textDf1)
+      .withColumn("translation", flatten(col("translation.translations")))
+      .withColumn("transliteration", col("translation.transliteration.text"))
+      .withColumn("translation", col("translation.text"))
+      .select("translation", "transliteration").collect()
+    assert(results.head.getSeq(0).mkString("\n") === "你好，你叫什么名字？")
+    assert(results.head.getSeq(1).mkString("\n") === "nǐ hǎo ， nǐ jiào shén me míng zì ？")
+  }
+
+  test("Translate to multiple languages") {
+    val results = translate
+      .setToLanguage(Seq("zh-Hans", "de"))
+      .transform(textDf1)
+      .withColumn("translation", flatten(col("translation.translations")))
+      .withColumn("translation", col("translation.text"))
+      .select("translation").collect()
+    val headStr = results.head.getSeq(0).mkString("\n")
+    assert(headStr === "你好，你叫什么名字？\nHallo, wie heißt du?")
+  }
+
+  test("Handle profanity") {
+    val results = translate
```

Comment:
when re-using the same estimator make translate a def not a lazy val. I know its weird but the setters actually MODIFY state globally which is a weird thing on SparkML's part

Reply:
Done. That solves my pain on creating multiple translators lol!